app/assets/javascripts/common/toggleScrollToPosition.js

```diff
@@ -0,0 +1,31 @@
+// Check for scroll position and scroll
+window.addEventListener("DOMContentLoaded", (event) => {
+  if (doesYPositionExistInLocalStorage()) {
+    scrollToYPosition();
+    removeYPositionLocalStorage();
+  }
+});
+
+// Create toggle change listener and add y position on 'change' event
+function createToggleListener(id) {
+  const toggleElement = document.getElementById(id);
+  toggleElement.addEventListener("change", (event) => {
+    setYPositionLocalStorage();
+  });
+}
+
+function setYPositionLocalStorage() {
+  localStorage.setItem("toggleScrollToPosition", window.pageYOffset);
```

Comment:
Can the key of the `localStorage` item be renamed to something more precise, like `scrollPosition`? That clearly conveys what we are storing. `toggleScrollToPosition` is a verb, so it makes it seem like we are storing an action of some kind.

Reply:
changed to `yScrollPosition`